### DISCUSSION TITLE
SocketException

### DISCUSSION THREAD

**Discussion by Vinayak006** [Imgkl/EventFlux#14] [Q&A] [answered]
When there is no internet, EventFlux throws SocketException, I don't know how to catch it or deal with it. I have tried using `try block` and also used the `onError` from connect method, but nothing helps. 

Any suggestions on how to handle it?

Thanks in advance.

**Comment by Imgkl** [comment, marked as answer]
Hey @Vinayak006 This is something people noticed and brought to my attention. 

I'm working on a fix. Will be out in few hours. 

FYI, https://github.com/Imgkl/EventFlux/issues/11

**Comment by Vinayak006** [comment]
Hey, Can you provide a sample code for catching ClientSocketException. After establishing a connection and network is turned off, it throws clientSocketException. I have tried try catch but didn't worked. Any Idea?

My current implementation:
```void _connect() {
    EventFlux.instance.connect(
      EventFluxConnectionType.get,
      header: <String, String>{
        "Authorization": token,
      },
      autoReconnect: true,
      AppConst.API_BASE_URL + ApiEndPoints.SSE,
      onSuccessCallback: (final EventFluxResponse? response) {
       // response handling
      },
    );
  }

**Reply by Imgkl** [reply]
Let me get back to you on this. 

Ideally any exceptions thrown by stream or if the status code is not 2xx. `onError` method will be triggered.
